GotExplorer.API/Controllers/AccountController.cs

```diff
@@ -61,12 +62,123 @@ public async Task<IActionResult> Login([FromBody] LoginDTO loginDTO)
             return Ok(result);
         }
 
-        // TODO: Delete endpoint.
+        /// <summary>
+        /// Get user data by id.
+        /// </summary>
+        /// <param name="id">user id</param>
+        /// <response code="200">Successfully retrieved user data</response>
+        /// <response code="400">The request data is invalid or incomplete.</response>
+        /// <response code="404">User not found.</response>
+        /// <response code="500">An unexpected error occurred on the server.</response>   
+        [HttpGet("get/{id}")]
+        [ProducesResponseType(typeof(GetUserDTO),200)]
+        [ProducesResponseType(typeof(ProblemDetails), 400)]
+        [ProducesResponseType(typeof(ProblemDetails), 404)]
+        [ProducesResponseType(typeof(ProblemDetails), 500)]
+        public async Task<IActionResult> GetUserById(string id)
+        {
+            var result = await _userService.GetUserById(id);
+            return Ok(result);
+        }
+
+        /// <summary>
+        /// Update user data by id stored in jwt claims. Require Authorization.
+        /// </summary>
+        /// <param name="updateUserDTO">The data to update the user's profile</param>
+        /// <response code="200">Successfully updated the user data.</response>
+        /// <response code="400">The request data is invalid or incomplete.</response>
+        /// <response code="404">User not found.</response>
+        /// <response code="500">An unexpected error occurred on the server.</response>        
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        [HttpPost("update")]
+        [ProducesResponseType(200)]
+        [ProducesResponseType(typeof(ProblemDetails), 400)]
+        [ProducesResponseType(typeof(ProblemDetails), 404)]
+        [ProducesResponseType(typeof(ProblemDetails), 500)]
+        public async Task<IActionResult> UpdateUserById([FromBody] UpdateUserDTO updateUserDTO)
+        {
+            updateUserDTO.Id = User.Claims.Where(x => x.Type == "Id").FirstOrDefault()?.Value;
```

Comment:
Move to custom or use existing extension
Do not use where when we need single record